components/eamxx/src/share/grid/remap/vertical_remapper.cpp

```diff
@@ -51,10 +53,32 @@ VerticalRemapper (const grid_ptr_type& src_grid,
   scorpio::register_file(map_file,scorpio::FileMode::Read);
   m_num_remap_levs = scorpio::get_dimlen_c2f(map_file.c_str(),"nlevs");
   scorpio::eam_pio_closefile(map_file);
-  auto tgt_grid = src_grid->clone("Point Grid",false);  //<-- ?? For Luca, why does the 1st arg have to be PointGrid?  Shouldn't the tgt_grid be of the same type as the src_grid if being cloned?  Note, choosing any other string threw an error for an "unsupported grid".
-  tgt_grid->reset_num_vertical_lev(m_num_remap_levs);
+
+  auto tgt_grid_gids = src_grid->get_unique_gids();
```

Comment:
I would add a comment here, explaining that this new grid has same 2d gids as src grid, and only differs in the number of levels.

Reply:
Btw, you could just do
```
auto tgt_grid = src_grid->clone("vertical_remap_tgt_grid", /*shallow = */ false);
tgt_grid->reset_num_vertical_lev  (m_num_remap_levs);
this->set_grids(src_grid,tgt_grid);
```
which is prob a bit more self-documenting.

Reply:
Actually, you could do a shallow copy, since all the grid fields (geo data, gids, ...) are the same, and you only need to change nlevs. This would save us some storage (the hard copy of geo data and gids).

Reply:
The `abstract_grid` class takes issue with both a shallow and deep copy in this case:
```
/ascldap/users/asdonah/sandbox_1/scream/components/eamxx/tests/uncoupled/shoc/shoc_standalone.cpp:29: FAILED:
  {Unknown expression after the reported line}
due to unexpected exception with message:
  /ascldap/users/asdonah/sandbox_1/scream/components/eamxx/src/share/grid/
  abstract_grid.cpp:167: FAIL:
  not has_geometry_data(name)
  Error! Cannot create geometry data, since it already exists.
    - grid name: vertical_remap_tgt_grid
    - geo data name: area
    - geo data layout: <COL>(433)
    - input layout: <COL>(433)
```

Reply:
if I follow the comment below then I hit a separate error:
```
/ascldap/users/asdonah/sandbox_1/scream/components/eamxx/tests/uncoupled/shoc/shoc_standalone.cpp:29: FAILED:
  {Unknown expression after the reported line}
due to unexpected exception with message:
  /ascldap/users/asdonah/sandbox_1/scream/components/eamxx/src/share/grid/
  remap/abstract_remapper.hpp:130: FAIL:
  src_fid.get_grid_name()==m_src_grid->name()
  Error! Input FieldIdentifier has the wrong grid name:
     - input src fid grid name: Point Grid
     - remapper src grid name:  vertical_remap_tgt_grid
```

Reply:
Ah, right. Would be nice to add `void Field::share_data (const Field& src)` to circumvent this problem.